File: ruby/problem_006.rb




sum_of_squares = (1..100).map {|s| s * s}.inject(0) {|s, x| s + x}
square_of_sum  = (1..100).inject(0) {|s, x| s + x} ** 2

p square_of_sum - sum_of_squares

File: python/problem_006.py
#!/usr/bin/python
#
# http://projecteuler.net/index.php?section=problems&id=6

import math

def square_of_sum(n):
    return ((n*(n+1))//2)**2

def sum_of_squares(n):
    return ((n*(n+1)*(2*n + 1))//6)

n = 100
print abs(sum_of_squares(n) - square_of_sum(n))
